src/webgpu/capability_info.ts

```diff
@@ -87,135 +87,1083 @@ export const kGeneratableErrorScopeFilters = kErrorScopeFilters.filter(
-  'depth32float':          [            ,              ,          ,        ,    true,     false,          ,      true,     false,              'depth',               4],
-  'depth16unorm':          [            ,              ,          ,        ,    true,     false,          ,      true,      true,              'depth',               2],
-  'stencil8':              [            ,              ,          ,        ,   false,      true,          ,      true,      true,               'uint',               1],
+/**
+ * Defaults applied to all tables automatically. Used only inside `tableWithDefaults`.
+ * This ensures keys are never missing, always explicitly `undefined`.
+ *
+ * All top-level keys must be defined here, or they won't be exposed.
+ */
+const kUniversalDefaults = {
+  blockWidth: undefined,
+  blockHeight: undefined,
+  color: undefined,
+  depth: undefined,
+  stencil: undefined,
+  colorRender: undefined,
+  multisample: undefined,
+  feature: undefined,
+  baseFormat: undefined,
+
+  sampleType: undefined,
+  copySrc: undefined,
+  copyDst: undefined,
+  bytesPerBlock: undefined,
+  renderable: false,
+  renderTargetPixelByteCost: undefined,
+  renderTargetComponentAlignment: undefined,
+
+  // IMPORTANT:
+  // Add new top-level keys both here and in TextureFormatInfo_TypeCheck.
+} as const;
+/**
+ * Takes `table` and applies `defaults` to every row, i.e. for each row,
+ * `{ ... kUniversalDefaults, ...defaults, ...row }`.
+ * This only operates at the first level; it doesn't support defaults in nested objects.
+ */
+function tableWithDefaults<Defaults extends {}, Table extends { readonly [K: string]: {} }>({
+  defaults,
+  table,
+}: {
+  defaults: Defaults;
+  table: Table;
+}): {
+  readonly [F in keyof Table]: {
+    readonly [K in keyof typeof kUniversalDefaults]: K extends keyof Table[F]
+      ? Table[F][K]
+      : K extends keyof Defaults
+      ? Defaults[K]
+      : typeof kUniversalDefaults[K];
+  };
+} {
+  return Object.fromEntries(
+    Object.entries(table).map(([k, row]) => [k, { ...kUniversalDefaults, ...defaults, ...row }])
+    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  ) as any;
+}
+
+/** "plain color formats", plus rgb9e5ufloat. */
+const kRegularTextureFormatInfo = tableWithDefaults({
+  defaults: { blockWidth: 1, blockHeight: 1 },
+  table: {
+    // plain, 8 bits per component
+
+    r8unorm: {
+      color: { type: 'float', copySrc: true, copyDst: true, storage: false, bytes: 1 },
+      colorRender: { blend: true, resolve: true, byteCost: 1, alignment: 1 },
+      renderable: true,
+      /*prettier-ignore*/ get renderTargetComponentAlignment() { return this.colorRender.alignment; },
+      /*prettier-ignore*/ get renderTargetPixelByteCost() { return this.colorRender.byteCost; },
+      multisample: true,
+      /*prettier-ignore*/ get sampleType() { return this.color.type; },
```

Comment:
Not sure if it's possible but the getters are repeated a lot (and clutter the table quite a bit...), idk if it's possible to make them conditionally defined depending on what sub-types are defined? i.e. if `color` is defined, then define `get sampleType() { return this.color.type; };`, otherwise don't define it? Not sure what is possible in TS though.

If it were possible, it would also make the deprecating part a lot easier since we would only have to delete the default parts (not the defs in every entry).

Feel free to leave these for now if it's not immediate obvious how to accomplish that though since I assume they are removed entirely in a follow up or something anyways.

Reply:
All of these getters are kind of hacked in. They're all for the deprecated fields so I didn't put much effort into them... just wrote a small block of code that I could copy paste into every format. They could be the same for every format but I didn't know how to / didn't want to figure out how to make copy getters from the default object into the final object - the tableWithDefaults thing actually calls all of the getters when the table is being constructed so they get baked into the object as actual values.

I'm just going to leave them until they're removed.